### PR TITLE
Bring back multiline comments

### PR DESCRIPTION
**Pull request by rullzer**
Fixes https://github.com/nextcloud/server/issues/942

Before:
before

After:
after

Which gives us multiline comments again! Whoohoo.
Only thing that we need to think about is how to handle the scroll bar (move to left?)

scroll

CC: @nickvergessen @nextcloud/designers @MorrisJobke

### CODE REVIEW

**Comment by eppfel** [comment]
Scrollbar on the left would be super weird.
Why is the button inside the text field anyway? With single line comments, you were able to send via `return`, so maybe the arrow signals this behavior. But for multiline the action should be separate. (to show some status inside the field seems right)

**Comment by eppfel** [comment]
I mean the cancel button is there, too.

File: core/js/core.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
 		"backbone/backbone.js",
 		"es6-promise/dist/es6-promise.js",
 		"davclient.js/lib/client.js",
-		"clipboard/dist/clipboard.js"
+		"clipboard/dist/clipboard.js",
+        "autosize/dist/autosize.min.js"
```
Comment (nickvergessen) [inline core/js/core.json]: tabs vs spaces

File: core/vendor/autosize/build.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+var pkg = require('./package.json');
```
Comment (nickvergessen) [inline core/vendor/autosize/build.js]: gitignore some files?

**Comment by rullzer** [comment]
@eppfel well I was following https://github.com/nextcloud/server/issues/942#issuecomment-242378602

But feel free to hijack to improve the design :)

**Comment by jancborchardt** [comment]
@rullzer @eppfel this field should be autosized, meaning there should _never_ ever be a scrollbar. :) Roeland, it seems you already added it?

**Comment by rullzer** [comment]
Okidoki now fully autosizing

Review time!

**Comment by MorrisJobke** [comment]
```
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Comments.CommentsTabView tests editing comments saves message and updates comment item when clicking save FAILED
	Expected Object({ message: 'New message' }) to equal Object({ message: 'modified message' }).
	/drone/src/github.com/nextcloud/server/apps/comments/tests/js/commentstabviewSpec.js:335:45
```

**Comment by rullzer** [comment]
Fixed

**Comment by nickvergessen** [comment]
Works 👍

**Comment by nickvergessen** [comment]
What I just noticed, when you have leading or trailing newlines/whitespace they are displayed in the comment, until you do a page refresh.
But I guess that is not related and should be fixed in a second PR?

**Comment by rullzer** [comment]
Yeah lets fix that separatly.

**Comment by MorrisJobke** [comment]
I will review this - there are only some layout glitches between firefox and Chrome.

File: apps/comments/css/comments.css [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 #commentsTabView .newCommentForm .submit {
 	position: absolute;
-	top: 1px;
+	bottom: 1px;
```
Comment (MorrisJobke) [inline apps/comments/css/comments.css]: With this it looks in Firefox good, but in Chrome not that good. Then I could change this to `5px` to look in Chrome good, but in Firefox not centered. The reason is that the surrounding element (a `<form>`) is in Chrome not fitted around the textarrea but a bit longer at the bottom.

bildschirmfoto von 2016-10-05 17-32-41

@nextcloud/designers Any idea how we could solve this?

Comment (MorrisJobke) [inline apps/comments/css/comments.css]: This is only 75 px in height in Firefox (and those are the 4 additional pixels that are needed in Chrome) :disappointed:

Comment (MorrisJobke) [inline apps/comments/css/comments.css]: This is how it looks without the change to `bottom: 5px` in chrome: 
bildschirmfoto von 2016-10-05 17-34-59

**Comment by Espina2** [comment]
What you think in making something like this. (Very quick mockup) 
Its very similar to What I did for the news page. 

comments 2
comments 1

File: apps/comments/css/comments.css (hunk 3 above)
Comment (skjnldsv) [inline apps/comments/css/comments.css]: @ChristophWurst @MorrisJobke You need to put a display:block on the textarea. :)
It should do the trick and the form should now take the full textarea size.

Comment (skjnldsv) [inline apps/comments/css/comments.css]: firefox firefox
chrome chrome

**Comment by rullzer** [comment]
@Espina2 o wow! I like it!
But lets first get this in and then change it :D

**Review by skjnldsv — APPROVED** [review]
👍

**Comment by MorrisJobke** [comment]
> But lets first get this in and then change it :D

Yep makes sense 👍 We should separate the different topics here.

**Comment by skjnldsv** [comment]
Always forgot the approval doesn't work on reviews! :expressionless:

LGTM 👍

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by jancborchardt** [comment]
@Espina2 very nice! Just two things:
- We don’t need the extra cointainer around the comment, whitespace is enough.
- Right now we sort comments with the newest up top, and the form is also on the top. This we should keep especially when we switch to introducing a timeline.
- In general, the introduction of a timeline necessitates that the layout of the comments and activites is in line with each other: https://github.com/nextcloud/server/issues/658 – maybe best work on a mockup for that and we can continue discussion there :) (cause the mockup from me there is already quite old and only crude)

**Comment by Espina2** [comment]
@jancborchardt Thanks dude. 

Yup, we should discuss this in you issue. This is only a quick sketch, for showing how can work not to much thinking in it. 👍

**Comment by MorrisJobke** [comment]
Only sign off check failed -> merge

**Pull request merged**